graql/reasoner/tree/ResolutionTree.java

```diff
@@ -66,20 +74,35 @@ public void clear(){
     }
 
     public void addState(ResolutionState state) {
-        ResolutionState parent = state.getParentState();
+        AnswerPropagatorState parent = state.getParentState();
         if (parent == null) return;
 
         if (state.isAnswerState()) {
-            Node parentNode = getNode(parent);
-            if (parentNode != null){
-                ConceptMap sub = state.getSubstitution();
-                parentNode.addAnswer(sub);
-            }
+            addAnswerState((AnswerState) state, parent);
         } else {
             addChildToNode(parent, state);
         }
     }
 
+    private void addAnswerState(AnswerState state, AnswerPropagatorState parent){
```

Comment:
This feels out of places, we shouldn't have to do much/any logic in the tree recording that is already being done elsewhere :)